local-macros/src/lib.rs

```diff
@@ -74,15 +74,100 @@ pub fn ast_type(_attr: TokenStream, item: TokenStream) -> TokenStream {
                 }
             }
         }
-        _ => panic!("Expected struct"),
+        Enum(DataEnum { variants, .. }) => {
+            let variant_names = variants.iter().map(|variant| &variant.ident);
+            let variant_names_2 = variant_names.clone();
```

Comment:
Apparently `quote!` is taking ownership of `variant_names` when I use it in a `#(...#variant_names...),*` so I made n copies of it